File: source/web/web.sph/App/partial/FeatureDefinition.js
/// <reference path="../schemas/sph.domain.g.js" />
/// <reference path="../../Scripts/jquery-2.0.3.intellisense.js" />
/// <reference path="../../Scripts/underscore.js" />
/// <reference path="../durandal/amd/require.js" />
var bespoke = bespoke || {};
bespoke.sph = bespoke.sph || {};
bespoke.sph.domain = bespoke.sph.domain || {};


bespoke.sph.domain.FeatureDefinitionPartial = function (model) {
    var getAvailableQuantityOptions = function () {
        var q = ko.observableArray();
        for (var i = 0; i <= model.AvailableQuantity(); i++) {
            q.push(i);
        }
        return q;
    };
    return {
        AvailableQuantityOptions: getAvailableQuantityOptions,
    };
};

File: source/domain/domain.sph/Business/FeatureDefinition.cs
using System.Collections.Generic;

namespace Bespoke.Sph.Domain
{
    partial class FeatureDefinition : DomainObject
    {

        public List<int> AvailableQuantityOptions
        {
            get
            {
                var q = new List<int>();
                for (int i = 0; i <= this.AvailableQuantity; i++)
                {
                    q.Add(i);
                };

                return q;
            }

        }
    }
}
